gcloud/pubsub/client.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from gcloud._helpers import get_default_project
+from gcloud._helpers import _ClientProxy
+from gcloud.pubsub._implicit_environ import _require_connection
+from gcloud.pubsub import api
+from gcloud.pubsub.subscription import Subscription
+from gcloud.pubsub.topic import Topic
+
+
+class Client(object):
+    """Wrap :mod:`gcloud.pubsub` API objects.
+
+    :type connection: :class:`gcloud.pubsub.connection.Connection` or None
+    :param connection: The configured connection. Defaults to one inferred
+                       from the environment.
+
+    :type project: str or None
+    :param connection: The configured project. Defaults to the value inferred
+                       from the environment.
+    """
+
+    def __init__(self, connection=None, project=None):
+        self.connection = _require_connection(connection)
+        if project is None:
+            project = get_default_project()
+        self.project = project
+
+    def topic(self, name):
```

Comment:
@tmatsuo didn't explain *why* we had to suppress any unnecessary API calls... He just said "please do it". I'd like to know where that's coming from, considering I'd like to hit the API when retrieving a topic so I know whether it exists or not... 

Boto in s3 (as a separate example) does this, and has a "validate" keyword that says "don't make an API call" (the default is "please make the API call"). For example

```python
bucket = conn.get_bucket('name')  # API request
bucket = conn.get_bucket('name', validate=False)  # No API request
```

In the search API draft I have, I just called this "check", but the pattern is similar: https://github.com/GoogleCloudPlatform/gcloud-python/pull/892/files#diff-056d3999e591db310404f19fbf639531R105

For comparison, gcloud-node let's you instantiate this (almost) the same way:

```javascript
// Create a new topic:
pubsub.createTopic('name', function(err, topic) {
  // Do something with the topic
});

// Get a topic that may or may not exist:
var topic = pubsub.topic('name', {autoCreate: true});
// Do something with topic.

// Delete an existing topic:
var topic = pubsub.topic('name');
topic.delete();

```